src/main/java/com/pokedex/springmvc/controller/UserController.java

```diff
@@ -0,0 +1,126 @@
+        if (user == null) {
+            System.out.println("User with id " + id + " not found");
+            return new ResponseEntity<User>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<User>(user, HttpStatus.OK);
+    }
+ 
+     
+     
+    //-------------------Create a User--------------------------------------------------------
+     
+    @RequestMapping(value = "/user/", method = RequestMethod.POST)
+    public ResponseEntity<Void> createUser(@RequestBody User user,    UriComponentsBuilder ucBuilder) {
+        System.out.println("Creating User " + user.getUsername());
+ 
+        if (userService.isUserExist(user)) {
+            System.out.println("A User with name " + user.getUsername() + " already exist");
+            return new ResponseEntity<Void>(HttpStatus.CONFLICT);
+        }
+ 
+        userService.saveUser(user);
+ 
+        HttpHeaders headers = new HttpHeaders();
+        headers.setLocation(ucBuilder.path("/user/{id}").buildAndExpand(user.getId()).toUri());
+        return new ResponseEntity<Void>(headers, HttpStatus.CREATED);
+    }
+ 
+    
+     
+    //------------------- Update a User --------------------------------------------------------
+     
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.PUT)
+    public ResponseEntity<User> updateUser(@PathVariable("id") long id, @RequestBody User user) {
+        System.out.println("Updating User " + id);
+         
+        User currentUser = userService.findById(id);
+         
+        if (currentUser==null) {
+            System.out.println("User with id " + id + " not found");
+            return new ResponseEntity<User>(HttpStatus.NOT_FOUND);
+        }
+ 
+        currentUser.setUsername(user.getUsername());
+        currentUser.setAddress(user.getAddress());
+        currentUser.setEmail(user.getEmail());
+         
+        userService.updateUser(currentUser);
+        return new ResponseEntity<User>(currentUser, HttpStatus.OK);
+    }
+ 
+    
+    
+    //------------------- Delete a User --------------------------------------------------------
+     
+    @RequestMapping(value = "/user/{id}", method = RequestMethod.DELETE)
+    public ResponseEntity<User> deleteUser(@PathVariable("id") long id) {
+        System.out.println("Fetching & Deleting User with id " + id);
+ 
+        User user = userService.findById(id);
+        if (user == null) {
+            System.out.println("Unable to delete. User with id " + id + " not found");
+            return new ResponseEntity<User>(HttpStatus.NOT_FOUND);
+        }
+ 
+        userService.deleteUserById(id);
+        return new ResponseEntity<User>(HttpStatus.NO_CONTENT);
+    }
+ 
+     
+    
+    //------------------- Delete All Users --------------------------------------------------------
+     
+    @RequestMapping(value = "/user/", method = RequestMethod.DELETE)
+    public ResponseEntity<User> deleteAllUsers() {
+        System.out.println("Deleting All Users");
```

Comment:
use sl4j for logging
[baeldung.com/slf4j-with-log4j2-logback](https://www.baeldung.com/slf4j-with-log4j2-logback)